From penberg@cs.helsinki.fi  Tue Apr 21 15:38:22 2009
From: Pekka Enberg <penberg@cs.helsinki.fi>
Date: Wed,  8 Apr 2009 11:51:22 +0300
Subject: Staging: w35und: merge rest of wbhal.c to phy_calibration.c
To: greg@kroah.com
Cc: Pavel Machek <pavel@ucw.cz>
Message-ID: <1239180682-22860-4-git-send-email-penberg@cs.helsinki.fi>


Impact: cleanup

The remaining functions are local to phy_calibration.c so move them
there and remove wbhal.c.

Acked-by: Pavel Machek <pavel@ucw.cz>
Signed-off-by: Pekka Enberg <penberg@cs.helsinki.fi>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/winbond/Makefile          |    1 -
 drivers/staging/winbond/phy_calibration.c |   19 +++++++++++++++++++
 drivers/staging/winbond/wbhal.c           |   20 --------------------
 drivers/staging/winbond/wbhal_f.h         |    4 ----
 4 files changed, 19 insertions(+), 25 deletions(-)

--- a/drivers/staging/winbond/Makefile
+++ b/drivers/staging/winbond/Makefile
@@ -7,7 +7,6 @@ w35und-objs :=			\
 	wb35reg.o		\
 	wb35rx.o		\
 	wb35tx.o		\
-	wbhal.o			\
 	wbusb.o			\
 
 
--- a/drivers/staging/winbond/phy_calibration.c
+++ b/drivers/staging/winbond/phy_calibration.c
@@ -340,6 +340,25 @@ void _sin_cos(s32 angle, s32 *sin, s32 *
     }
 }
 
+static unsigned char hal_get_dxx_reg(struct hw_data *pHwData, u16 number, u32 * pValue)
+{
+	if (number < 0x1000)
+		number += 0x1000;
+	return Wb35Reg_ReadSync(pHwData, number, pValue);
+}
+#define hw_get_dxx_reg( _A, _B, _C ) hal_get_dxx_reg( _A, _B, (u32 *)_C )
+
+static unsigned char hal_set_dxx_reg(struct hw_data *pHwData, u16 number, u32 value)
+{
+	unsigned char ret;
+
+	if (number < 0x1000)
+		number += 0x1000;
+	ret = Wb35Reg_WriteSync(pHwData, number, value);
+	return ret;
+}
+#define hw_set_dxx_reg( _A, _B, _C ) hal_set_dxx_reg( _A, _B, (u32)_C )
+
 
 void _reset_rx_cal(struct hw_data *phw_data)
 {
--- a/drivers/staging/winbond/wbhal.c
+++ /dev/null
@@ -1,20 +0,0 @@
-#include "sysdef.h"
-#include "wbhal_f.h"
-#include "wblinux_f.h"
-
-unsigned char hal_get_dxx_reg(struct hw_data *pHwData, u16 number, u32 * pValue)
-{
-	if (number < 0x1000)
-		number += 0x1000;
-	return Wb35Reg_ReadSync(pHwData, number, pValue);
-}
-
-unsigned char hal_set_dxx_reg(struct hw_data *pHwData, u16 number, u32 value)
-{
-	unsigned char ret;
-
-	if (number < 0x1000)
-		number += 0x1000;
-	ret = Wb35Reg_WriteSync(pHwData, number, value);
-	return ret;
-}
--- a/drivers/staging/winbond/wbhal_f.h
+++ b/drivers/staging/winbond/wbhal_f.h
@@ -52,8 +52,6 @@ u32 hal_get_bss_pk_cnt(  struct hw_data 
 
 #define PHY_DEBUG( msg, args... )
 
-unsigned char hal_get_dxx_reg(  struct hw_data * pHwData,  u16 number,  u32 * pValue );
-unsigned char hal_set_dxx_reg(  struct hw_data * pHwData,  u16 number,  u32 value );
 #define hal_get_time_count( _P )	(_P->time_count/10)	// return 100ms count
 #define hal_detect_error( _P )		(_P->WbUsb.DetectCount)
 
@@ -68,7 +66,5 @@ unsigned char hal_set_dxx_reg(  struct h
 #define hal_join_request_stop(_A)
 #define hw_get_cxx_reg( _A, _B, _C )
 #define hw_set_cxx_reg( _A, _B, _C )
-#define hw_get_dxx_reg( _A, _B, _C )	hal_get_dxx_reg( _A, _B, (u32 *)_C )
-#define hw_set_dxx_reg( _A, _B, _C )	hal_set_dxx_reg( _A, _B, (u32)_C )
 
 
